### PR TITLE
refactor(backend): 集群列表的访问入口支持多域名返回 #5013

### PR DESCRIPTION
**Pull request by Maxwell-416258**

### CODE REVIEW

**Comment by tencentblueking-adm** [comment]
[CLA assistant check](https://bk-cla.bktencent.com:443/TencentBlueKing/blueking-dbm?pullRequest=5023) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://bk-cla.bktencent.com:443/TencentBlueKing/blueking-dbm?pullRequest=5023) before we can accept your contribution.<br/><hr/>**leifxu** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://bk-cla.bktencent.com:443/check/TencentBlueKing/blueking-dbm?pullRequest=5023) it.</sub>

File: dbm-ui/backend/db_meta/models/cluster_entry.py [hunk 1]
```diff
@@ -75,10 +75,10 @@ def get_cluster_entry_map(cls, cluster_ids: List[int]) -> Dict[int, Dict[str, st
                 continue
 
             # DNS 需额外区分主域名和从域名， entry 中 cluster.immute_domain 为主域名
-            # 那么不等于 cluster.immute_domain 的则理解为是从域名
+            # 如果entry.role为slave_entry，则为从域名
             if access_entry == entry.cluster.immute_domain:
                 cluster_entry_map[entry.cluster_id]["master_domain"] = access_entry
-            else:
+            elif entry.role == "slave_entry":
```
Comment (zhangzhw8) [inline dbm-ui/backend/db_meta/models/cluster_entry.py]: "slave_entry" -> ClusterEntryRole.SLAVE_ENTRY

**Review by zhangzhw8 — CHANGES_REQUESTED** [review]
1. 这个字段所有集群都有，因此不要在每个 resouces 里面添加，而是在 base 里面统一添加2. 现在这个实现方式，会导致 n+1 查询，不要在 to_cluster_representation 里加这里逻辑，改成在列表返回中统一添加

File: dbm-ui/backend/db_meta/models/cluster_entry.py (hunk 1 above)
Comment (Maxwell-416258) [reply dbm-ui/backend/db_meta/models/cluster_entry.py]: 好的，已改正

File: dbm-ui/backend/db_services/dbbase/resources/query.py [hunk 2]
```diff
@@ -496,6 +497,9 @@ def _filter_cluster_hook(
         for cluster in cluster_queryset:
             cluster_info = cls._to_cluster_representation(
                 cluster=cluster,
+                cluster_entry=[
+                    {"cluster_entry_type": entry.cluster_entry_type, "entry": entry.entry} for entry in cluster.entries
```
Comment (iSecloud) [inline dbm-ui/backend/db_services/dbbase/resources/query.py]: 这个加上entry的角色吧，这样前端才能知道哪些是主域名，哪些是从域名

Comment (Maxwell-416258) [reply dbm-ui/backend/db_services/dbbase/resources/query.py]: 已改

**Review by zhangzhw8 — APPROVED** [review]

**Review by zhangzhw8 — APPROVED** [review]

**Pull request merged**